### PR TITLE
Introduce metadatamanager for connector

### PR DESCRIPTION
**Pull request by caneGuy**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Related with #5062 #5671

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataManager.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import org.apache.logging.log4j.Logger;
+
+import javax.annotation.concurrent.GuardedBy;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class MetadataManager {
+    private static final Logger LOG = LogManager.getLogger(MetadataManager.class);
+    @GuardedBy("this")
+    private final ConcurrentHashMap<String, ConnectorMetadata> connectorsMetadata = new ConcurrentHashMap<>();
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/server/MetadataManager.java:16]: connectorMetadatas

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataManager.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    public void addMetadata(String catalogName, ConnectorMetadata metadata) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/server/MetadataManager.java:18]: synchronized?

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataManager.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        connectorsMetadata.put(catalogName, metadata);
+    }
+
+    public void removeMetadata(String catalogName) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/server/MetadataManager.java:22]: ditto

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataManager.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        connectorsMetadata.remove(catalogName);
+    }
+
+    // get metadata by catalog name
+    private Optional<ConnectorMetadata> getOptionalMetadata(String catalogName) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/server/MetadataManager.java:27]: ditto

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataManager.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+import com.starrocks.connector.ConnectorMetadata;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.annotation.concurrent.GuardedBy;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class MetadataManager {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/MetadataManager.java:13]: named MetadataMgr?

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataManager.java (hunk 2 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/server/MetadataManager.java:18]: ConcurrentHashMap can guarantee thread safety.

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/server/MetadataManager.java:18]: ConcurrentHashMap can guarantee thread safety

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+    public void addMetadata(String catalogName, ConnectorMetadata metadata) {
+        connectorMetadatas.put(catalogName, metadata);
+    }
+
+    public void removeMetadata(String catalogName) {
+        connectorMetadatas.remove(catalogName);
+    }
+
+    // get metadata by catalog name
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java:26]: LGTM

**Review by stephen-shelby — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 8 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|0|8|00.00%|

**Review by DorianZheng — DISMISSED** [review]
LGTM

**Review by stephen-shelby — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+import org.apache.logging.log4j.Logger;
+
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.annotation.concurrent.GuardedBy;
+
+public class MetadataMgr {
+    private static final Logger LOG = LogManager.getLogger(MetadataMgr.class);
+
+    @GuardedBy("this")
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java:16]: What is this annotation for?

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+import com.starrocks.connector.ConnectorMetadata;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.annotation.concurrent.GuardedBy;
+
+public class MetadataMgr {
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java:13]: Have we agreed to use `Mgr`? How about full name `Manager`?

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java (hunk 7 above)
Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java:16]: why not `synchonirzed`?

**Review by dirtysalt — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java (hunk 8 above)
Comment (caneGuy) [reply fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java:13]: this is just to keep the style consistent

**Review by dirtysalt — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 8 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|0|8|00.00%|

**Review by Youngwb — APPROVED** [review]

**Pull request merged**